### PR TITLE
Do not implicitly make transactions

### LINKED ISSUE #16: Address "Alternative return types" anti-pattern usage
### Possible Solutions

1. Remove action and always return the prepared parameters. User needs to manually call Ethers.call, Ethers.send or Ethers.estimate_gas with the result from the contract functions. The result would look like below. One downside would be losing the ability to have default actions.
```elixir
iex> ERC20.balance_of("0xADDRESS_HERE") |> Ethers.call()
{:ok, [1000]}
```

  2.  Introduce a Result struct which can hold a transaction hash and the return values at the same time.
```elixir
iex> ERC20.balance_of("0xADDRESS_HERE")
{:ok, %Result{return_values: [10000], tx_hash: nil}}
```

  4.  Move action out of the opts keyword into a separate function argument with a default action.
```elixir
iex> ERC20.balance_of("0xADDRESS_HERE")
{:ok, [1000]}

iex> ERC20.balance_of("0xADDRESS_HERE", :prepare)
{:ok, %{data: "0x70a08231...", selector: %ABI.FunctionSelector{...}, to: "0xCONTRACT"}
```

### PR DESCRIPTION
**Pull request by alisinabh**
To address the Alternative return types and also to not implicitly perform transactions on the blockchain, a solution is to require explicit calls to `Ethers.call`, `Ethers.send` or `Ethers.estimate_gas` functions by the user.

Now calling a function looks like this

```elixir
iex> Ethers.Contracts.ERC20.name() |> Ethers.call(to: "0xCONTRACT_ADDRESS")
{:ok, "WETH"}
```

### Tasks
- [x] Move call, send and estimate_gas to Ethers module
- [x] Remove perform_action function
- [x] Adapt tests
- [x] Update documentation generated for contracts
- [x] Update README.md

Closes #16

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -75,31 +75,28 @@ The documentation is also available giving the developer a first-class experienc
...
+State mutability: view
 
 ## Parameters
 
   • _owner: `:address`
-  • overrides: Overrides and options for the call. See Execution Options in
-    Ethers.Contract.
 
-## Return Types
+## Return Types (when called with `Ethers.send/2`)
```
Comment (NduatiK) [inline README.md:99]: Should this be `Ethers.call`?

File: lib/ethers.ex [hunk 2]
```diff
@@ -101,6 +118,159 @@ defmodule Ethers do
     end
   end
 
+  @doc """
+  Makes an eth_call to with the given data and overrides, Than parses
```
Comment (NduatiK) [inline lib/ethers.ex:122]: ~Than~ then

File: lib/ethers.ex [hunk 3]
```diff
@@ -101,6 +118,159 @@ defmodule Ethers do
...
+  def call!(params, overrides \\ []) do
+    case call(params, overrides) do
+      {:ok, result} -> result
+      {:error, reason} -> raise ExecutionError, reason
+    end
+  end
+
+  @doc """
+  Makes an eth_send rpc call to with the given data and overrides, Then returns the
+  transaction binary.
```
Comment (NduatiK) [inline lib/ethers.ex:186]: "transaction ~binary~ hash" might be clearer

**Review by NduatiK — COMMENTED** [review]
I love this change!

💙💙💙

 Here's a couple of documentation nitpicks

**Comment by alisinabh** [comment]
@NduatiK Nice catch. Thank you very much 💚

**Pull request merged**